### PR TITLE
Add a lat-lon baroclinic fluid example with immersed boundaries [WIP]

### PR DESCRIPTION
**Pull request by sandreza**
The following PR adds a new script with immersed boundaries on a lat-lon grid as a validation test

### CODE REVIEW

**Comment by francispoulin** [comment]
Thanks @sandreza for making this PR. 

Is there an image or animation you can share to help us visualize what this is doing?

**Comment by sandreza** [comment]
@francispoulin once it compiles I'll add a video :)

**Comment by glwagner** [comment]
We want this example to be baroclinic (varying with depth), right? I think it'd be nice if this experiment has both a coastline and undulating depth.

File: validation/immersed_boundaries/immersed_lat_lon_barotropic.jl [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+latitude = (-30, 30)
+Δφ = latitude[2] - latitude[1]
+
+resolution = 4 # degree
+Nx = round(Int, 360 / resolution)
+Ny = round(Int, Δφ / resolution)
+Nz = 10
+
+# A spherical domain
+underlying_grid = RegularLatitudeLongitudeGrid(size = (Nx, Ny, Nz),
```
Comment (glwagner) [inline validation/immersed_boundaries/immersed_lat_lon_barotropic.jl:46]: Looks like there are a few points in z, so its just the name of the PR that has to change!

**Comment by glwagner** [comment]
> @francispoulin once it compiles I'll add a video :)

Let's see that video!

**Comment by glwagner** [comment]
Any progress on this @sandreza ?

**Comment by glwagner** [comment]
Closing with @sandreza's permission.

**Pull request closed without merging**